clients/adapter/src/main/java/org/eclipse/hono/adapter/client/command/CommandContext.java

```diff
@@ -59,17 +61,18 @@
     /**
      * Indicates to the sender that the command message could not be delivered to its target due to
      * reasons that are not the responsibility of the sender of the command.
+     *
+     * @param error The delivery error. May be {@code null}.
      */
-    void release();
+    void release(Throwable error);
```

Comment:
I'd rather keep the existing variant without a param and instead overload the method. Explicitly passing in a `null` argument into methods doesn't look too good FMPOV, in particular when the method has a single parameter only ...

Reply:
I've added the extra `release()` method.